samplers/src/test/java/io/opentelemetry/contrib/sampler/CelBasedSamplerTest.java

```diff
@@ -0,0 +1,275 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.sampler;
+
+import static io.opentelemetry.semconv.HttpAttributes.HTTP_RESPONSE_HEADER;
+import static io.opentelemetry.semconv.UrlAttributes.URL_FULL;
+import static io.opentelemetry.semconv.UrlAttributes.URL_PATH;
+import static io.opentelemetry.semconv.incubating.ThreadIncubatingAttributes.THREAD_NAME;
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.clearInvocations;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import dev.cel.common.CelValidationException;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.IdGenerator;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingDecision;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CelBasedSamplerTest {
+  private static final String SPAN_NAME = "MySpanName";
+  private static final SpanKind SPAN_KIND = SpanKind.SERVER;
+  private final IdGenerator idsGenerator = IdGenerator.random();
+  private final String traceId = idsGenerator.generateTraceId();
+  private final String parentSpanId = idsGenerator.generateSpanId();
+  private final SpanContext sampledSpanContext =
+      SpanContext.create(traceId, parentSpanId, TraceFlags.getSampled(), TraceState.getDefault());
+  private final Context parentContext = Context.root().with(Span.wrap(sampledSpanContext));
+
+  private final List<CelBasedSamplingExpression> expressions = new ArrayList<>();
+
+  @Mock(strictness = Mock.Strictness.LENIENT)
+  private Sampler delegate;
+
+  @BeforeEach
+  public void setup() throws CelValidationException {
+    when(delegate.shouldSample(any(), any(), any(), any(), any(), any()))
+        .thenReturn(SamplingResult.create(SamplingDecision.RECORD_AND_SAMPLE));
+
+    expressions.add(
+        new CelBasedSamplingExpression(
+            CelBasedSampler.celCompiler
+                .compile(
+                    "spanKind == 'SERVER' && attribute[\""
+                        + URL_FULL
```

Comment:
This works because `AttributeKey.toString()` merely returns the key (String), but if the `toString()` one day had type information in it or something, this could fail. It can be made slightly more explicit/robust by using `URL_FULL.getKey()`.

Reply:
Good catch. Is fixed now.